File: ios/CoursesWidget/Course.swift
//
//  Course.swift
//  CoursesWidgetExtension
//
//  Created by Yazeed AlKhalaf on 11/03/2023.
//

import Foundation

struct Course: Identifiable, Decodable, Hashable {
    let id = UUID()
    
    let courseCode: String
    let roomName: String
    let startTime: String
    let endTime: String
}

File: mobile/android/app/src/main/java/dev/alkhalaf/alyamamah/models/Course.kt
package dev.alkhalaf.alyamamah.models

import java.util.UUID
import android.graphics.Color
import org.json.JSONObject

data class Course(
        val id: String = UUID.randomUUID().toString(),
        val color: Int,
        val courseCode: String,
        val roomName: String,
        val startTime: TimeOfDay,
        val endTime: TimeOfDay
) {
    companion object {
        fun fromJson(json: JSONObject): Course {
            val color = Color.parseColor(json.getString("color"))
            val courseCode = json.getString("courseCode")
            val roomName = json.getString("roomName")
            val startTime = TimeOfDay.fromJson(json.getJSONObject("startTime"))
            val endTime = TimeOfDay.fromJson(json.getJSONObject("endTime"))

            return Course(
                    color = color,
                    courseCode = courseCode,
                    roomName = roomName,
                    startTime = startTime,
                    endTime = endTime
            )
        }
    }
}
